### DISCUSSION TITLE
Add temperature preset entities and redesign the configuration process

### DISCUSSION THREAD

**Discussion by jmcollin78** [jmcollin78/versatile_thermostat#410] [Announcements]
## What's Changed
* Make the temperature preset as persistent entities / redesign of the configuration menu by @jmcollin78 in https://github.com/jmcollin78/versatile_thermostat/pull/409

**Full Changelog**: https://github.com/jmcollin78/versatile_thermostat/compare/5.4.2...6.0.0

---
# Changes in version 6.0

## Temperature entities for presets
Preset temperatures are now directly accessible in the form of entities linked to VTherm.
Example :

![Temperature entities](https://github.com/jmcollin78/versatile_thermostat/blob/main/images/temp-entities-1.png)

The Boost, Comfort, Eco and Frost Protection entities allow you to directly adjust the temperatures of these presets without having to reconfigure the VTHerm in the configuration screens.
These modifications persist after a restart and are taken into account immediately by VTherm.

Depending on the functions activated, the list of temperatures may be more or less complete:
1. If presence management is activated, absence presets are created. They are suffixed with 'abs' for absence,
2. If air conditioning management (AC Mode) is activated, air conditioning mode presets are created. They are suffixed with 'clim' for air conditioning. Only the Frost protection preset has no equivalent in air conditioning mode,
3. The different absent and air conditioning combinations can be created depending on the configuration of the VTherm

If a VTherm uses the presets of the central configuration, these entities are not created, because the temperatures of the presets are managed by the central configuration.

### In the case of a central configuration
If you have configured a central configuration, this also has its own presets which meet the same rules as stated above.
Example of a central configuration with presence management and AC (air conditioning) mode:

![Temperature entities](https://github.com/jmcollin78/versatile_thermostat/blob/main/images/temp-entities-2.png)

In the case of a change of a central configuration temperature, all VTherms that use this preset are immediately updated.

## Redesign of the configuration menu
The configuration menu has been completely revised. It dynamically adapts to the user's choices and allows direct access to the settings of the desired function without having to go through the entire configuration tunnel.

To create a new VTherm, you will first need to choose the type of VTherm:

![VTherm choice](https://github.com/jmcollin78/versatile_thermostat/blob/main/images/config-main0.png)

Then, you now access the following configuration menu:

![VTherm menu](https://github.com/jmcollin78/versatile_thermostat/blob/main/images/config-menu.png)

Each part to be configured is directly accessible, without having to go through the entire configuration tunnel as before.

You will note the menu option named `Functions` which allows you to choose which functions will be implemented for this VTherm:

![VTherm features](https://github.com/jmcollin78/versatile_thermostat/blob/main/images/config-features.png)

Depending on your choices, the main menu will adapt to add the necessary options.

Example of menu with all functions checked:

![VTherm menu](https://github.com/jmcollin78/versatile_thermostat/blob/main/images/config-menu-all-options.png)
You can see that the 'Opening detection', 'Motion detection', 'Power management' and 'Presence management' options have been added. You can then configure them.

### The 'Incomplete configuration' and 'Finalize' menu options

The last menu option is special. It allows you to validate the creation of the VTherm when all the functions have been correctly configured.
If one option is not configured correctly, the last option is:

![Incomplete configuration](https://github.com/jmcollin78/versatile_thermostat/blob/main/images/config-not-complete.png)

Its selection does nothing but prevents you from finalizing the creation (resp. modification) of the VTherm.
**You must then search in the options which one is missing**.

Once all configuration is valid, the last option changes to:

![Complete configuration](https://github.com/jmcollin78/versatile_thermostat/blob/main/images/config-complete.png)

Click on this option to create (resp. modify) the VTherm:

![Configuration Complete](https://github.com/jmcollin78/versatile_thermostat/blob/main/images/config-terminate.png)

<hr /><em>This discussion was created from the release <a href='https://github.com/jmcollin78/versatile_thermostat/releases/tag/6.0.0'>Add temperature preset entities and redesign the configuration process</a>.</em>

**Comment by maia** [comment]
This is an awesome change. But please, use english screenshots in all documentation in the future. An english text with french screenshots makes it very hard to understand. Thanks!

**Reply by jmcollin78** [reply]
My HA is configured in French. As so my screen copy are in French. I will try to change this for the future

**Comment by MBRetrofit** [comment]
'
Use motion detection is always ticked, even if I start everything from the scratch
Use a central boiler still does not show any values after all Windows, Android, Home assistant app, etc restarted
![Central](https://github.com/jmcollin78/versatile_thermostat/assets/151366453/53d43329-e879-4c3e-b948-84bf8324daec)
![Motion](https://github.com/jmcollin78/versatile_thermostat/assets/151366453/8e44f87e-d40b-4ba1-b8ce-c3964ad00bdf)

**Reply by jmcollin78** [reply]
You have to clear the cache. In the companion app it is hidden in a developement menu. But I never remember where

**Reply by MBRetrofit** [reply]
![Cache](https://github.com/jmcollin78/versatile_thermostat/assets/151366453/72568040-f809-4a14-96b5-9f0b0d4ea629)
In Chrome enable Dev Tools then right click on Refresh button and the Delete Cache and Hard reset will be available. All done few times.

Motion Detection is always enabled
Central Boiler settings is empty.

**Reply by MBRetrofit** [reply]
Same issue at today new version

**Reply by jmcollin78** [reply]
It seems that you are alone with this problem. Don't kown why but it is certainly related to your environment. I speak about the translation that are missing.

**Reply by MBRetrofit** [reply]
I did not catch your "translation that are missing" issue, but the "Use Motion Detection" is ticked by default, if I try un-tick, save configuration and check again, the Use Motion Configuration is still ticked.

My main problem is more complicated now that need to be solved.
I just noticed that my Heating Automation stop working, as maybe the entity related to idle, hot, etc was renamed or changed.

type object 'HVACMode' has no attribute 'HEATING'

**Reply by MBRetrofit** [reply]
All attributes solved after latest update. Thank you!

**Comment by mleiber** [comment]
For the new way of handling presets, am I supposed to still see the presets in the configuration menu ? The main menu has a `Presets` entry, clicking that leads to a dialog asking if to use the central configuration, leaving that unchecked and clicking Submit brings you back to the main menu. Shouldn't that open a dialogue to enter the preset temperatures?
Or is the only way now to alter the presets by creating a screen myself with the number.* entries on it?
One or the other way, great integraton that simplifies handling of heating control substantially.

**Reply by jmcollin78** [reply]
> For the new way of handling presets, am I supposed to still see the presets in the configuration menu ?

No. With V6.0 the preset temperature have to be configured in each entity. That is the main feature of this release.

> Or is the only way now to alter the presets by creating a screen myself with the number.* entries on it?

Yes, that is an option. The other option is to go directly in the device configuration and change the number entities.

**Comment by tbrasser** [comment]
It's the first time using this component. So I don't know how it worked before.

But if I select motion then my vtherm should go into some preset right?

It stays off:

![Screenshot_2024-03-19-10-13-11-583_io.homeassistant.companion.android.jpg](https://github.com/jmcollin78/versatile_thermostat/assets/2816560/0c2b7c42-2661-4439-a3f6-dac2bf69dd5b)

**Reply by jmcollin78** [reply]
Can you please open a new discussion because it has nothing to do with the subject (the new release).

Thank you in advance.

And I could suggest to read the README about the motion management (which is different but confusing with presence management). Typically you have to put the VTherm into the activity preset (which should be preset) for motion management to work.

**Comment by sleveille4** [comment]
since i updated to new release, i now have some warning in the log upon restart of HA:

```
This error originated from a custom integration.

Logger: custom_components.versatile_thermostat.number
Source: custom_components/versatile_thermostat/number.py:115
integration: Versatile Thermostat configuration (documentation, issues)
First occurred: 12:45:05 (8 occurrences)
Last logged: 12:45:07

Chambre1_elec - configuring Number non central, non AC, non AWAY for preset boost_temp
Sdb_elec - configuring Number non central, non AC, non AWAY for preset frost_temp
Sdb_elec - configuring Number non central, non AC, non AWAY for preset eco_temp
Sdb_elec - configuring Number non central, non AC, non AWAY for preset comfort_temp
Sdb_elec - configuring Number non central, non AC, non AWAY for preset boost_temp
```

**Reply by jmcollin78** [reply]
This have been fixed in 6.0.2 release.

**Reply by sleveille4** [reply]
issue no longer happening, thanks :)
